spanner/read.go

```diff
@@ -121,6 +130,29 @@ type RowIterator struct {
 	sawStats         bool
 }
 
+// RowsReturned returns, a lower bound on the number of rows returned by the query.
+// Currently, this requires the query to be executed with query stats enabled.
```

Comment:
We should also document that the query stats are only included in the last response of the gRPC stream. That means that if you run a query with query stats enabled that returns a large number of rows, then this method is only guaranteed to return the number of rows once you have iterated through all the rows (which again makes it a bit less useful). So maybe add the following to this sentence: `The query stats are only guaranteed to be available after iterating through all the rows.`

And change the last point to something like this:

```
// If the query was executed without query stats enabled, if the query stats have not yet been
// returned by the server (the query stats are included in the last message of the query stream),
// or if it is otherwise impossible to determine the number of rows in the resultset, -1 is returned.
```

Reply:
>We should also document that the query stats are only included in the last response of the gRPC stream. That means that if you run a query with query stats enabled that returns a large number of rows, then this method is only guaranteed to return the number of rows once you have iterated through all the rows (which again makes it a bit less useful). So maybe add the following to this sentence: The query stats are only guaranteed to be available after iterating through all the rows.

This is only true for DML/PDML queries, for normal execute SQL queries I don't see that restriction, and usecases of decoding from DML/PDML returned rows is usually with Returning clause.

Reply:
Removed the logic to rely on ReturnedRows stats since for large dataset row count is returned with last result set only hence it won't help in preallocation.
Thanks